### ISSUE TITLE
Select specific column when join two tables

### ISSUE THREAD

**Issue by mistraminayasa** [go-gorm/gorm#1815]
Why join in gorm always select all column? Can we select specific column when join two tables?

**Comment by s-takehana** [comment]
http://gorm.io/docs/query.html#Joins

```go
db.Table("users").Select("users.name, emails.email").Joins("left join emails on emails.user_id = users.id").Scan(&results)
```

**Comment by mistraminayasa** [comment]
Thanks, it works as I expected

**Comment by tfmertz** [comment]
Just figured out how to do this. Maybe there's a simpler way, but this is what worked for me:
```go
// Include all your columns here
selectItems := "users.id, users.first_name, users.last_name, roles.name"
db := models.DB.Table("users").Select(selectItems).
    Joins("JOIN roles ON users.role_id = roles.id")

rows, err := db.Rows()
if err != nil {
    return err
}

var users []User
for rows.Next() {
    var id uint32
    var firstName string
    var lastName string
    var roleName string // should hold your other tables value
     rows.Scan(&id, &firstName, &lastName, &roleName)
     user := User{
        Id: id,
        FirstName: firstName,
        LastName: lastName,
    }
    user.Role.Name = roleName

    // Should hold all the values on the users in a list
    users = append(users, user)
}
```

It's kind of an eyesore and hard to follow especially if your table has a ton of columns like mine, so if anyone has a better way, would love to hear it!

**Comment by xland** [comment]
Is there anyone has a better way?

**Comment by happyxhw** [comment]
I am despaired!
I suggest that gorm removes the function of join!
Is there anyone has a better way?

**Comment by shubhamgoyal41** [comment]
@tfmertz Thanks! This helped

**Comment by dumindu** [comment]
For me, this works. Only one column was needed in my case btw.
```
record := &struct {ID uint}{}
if err := db.Table("users").
	Select("users.id").
	Joins("join emails on emails.user_id = users.id").
	Where("emails.name = ? AND users.name = ?", emailName, username).
	Scan(record).Error; err != nil {
	return nil, err
}

return record, nil
```

**Comment by caodj** [comment]
Try this way
```
	// Include all your columns here
	var UserResult[] struct{
		Id uint32
		FirstName string
		LastName string
		RoleName string
	}
	selectItems := "users.id, users.first_name, users.last_name, roles.name"
	err = models.DB.Table("users").Select(selectItems).
		Joins("JOIN roles ON users.role_id = roles.id").Find(&UserResult).Error

	if err != nil {
		return err
	}
```
